### PR TITLE
User interface for Tic Tac Toe game

### PR DESCRIPTION
**Pull request by Maclenn77**
We added the flow of input and output in the main.rb. These are the features:

- A welcome message.
- The input of players' names.
- Selecting first move
- Displaying the board
- Switching turns
- Displaying result of the game and score
- Asking users if they want to play again.

We want to ask if we need to add the bin folders also in this branch and put the main.rb file inside that folder

**Thanks for reviewing!**

### CODE REVIEW

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, Just few things to work on. 
### General
- [x]  Yes you are to add  `lib` & `bin` directory from milestone 1
### Note: the Basic game flow is :
* [x]  game tells which player turn it is
* [x]  game asks player to select from available moves
* [x]  game informs player if selected move is invalid
* [x]  game displays board after player move
* [x]  game informs player if selected move is a winning move
* [x]  game informs player if selected move is a draw move
* [x]  game repeats all actions for next player's move
- [x] A loop is needed in this game
- [x] Inside loop, you  should  call actions repeated for each move
- [x] Your loop needs to be terminated at some point , It can be done with a variable .
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: bin/main [hunk 1]
```diff
@@ -1,2 +1,78 @@
...
+  hash[move] = 'X'
+  display_board(hash)
+  puts "#{player_two} it's your turn. Choose one number between 1 and 9."
+end
+
+# Loop
+def loop_game(name_one, name_two, hash)
+  count = 0
+
+  until count == 4
```
Comment (mcrd25) [inline bin/main]: Your game loop should be based on a condition that person won or lose, while you are technically using a variable to terminate the loop. This is an out of nowhere terminator.

Please remember that this main file does not have to run (Can even be done with comments), it's simply a way ensuring that you thought about everything listed in the previous review.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changed Required 🔍 👀 

Hey @Maclenn77! Hey @RossielCS! 👋 

Great job implementing the issues outlined from the previous review. 😄 
I see only one more thing to address! Check out my review comment. 

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya Douglas](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

File: bin/main.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
+hash = { 1 => 1, 2 => 2, 3 => 3, 4 => 4, 5 => 5, 6 => 6, 7 => 7, 8 => 8, 9 => 9 }
```
Comment (mcrd25) [inline bin/main.rb]: [OPTIONAL]

Since you already have the main file this file is sort of redundant. It's not necessary.

**Review by mosesogwo — APPROVED** [review]
**HI @Maclenn77, @RossielCS:clap:. You can now move on to the next milestone:+1:**

### Status: Approved:heavy_check_mark:

**Pull request merged**